### ISSUE TITLE
In HttpRequestFirstLine::HttpRequestFirstLine there might be an undeflow for a bad packet

### ISSUE THREAD

**Issue by amritanshujohri** [seladb/PcapPlusPlus#543]
```
m_Method = parseMethod((char*)m_HttpRequest->m_Data, m_HttpRequest->getDataLen());
 if (m_Method == HttpRequestLayer::HttpMethodUnknown)
 {
 m_UriOffset = -1;
 LOG_DEBUG("Couldn't resolve HTTP request method");
 }
 else
 m_UriOffset = MethodEnumToString[m_Method].length() + 1;
 parseVersion(); <<---in parse version we can be off by 1 if m_UriOffset = -1, that can result in an underflow.
```

**Comment by seladb** [comment]
Yes, this is correct. We should probably check inside of `parseVersion()` if `m_UriOffset != -1`. Would you consider opening a PR with the fix?

**Comment by amritanshujohri** [comment]
I will testing it locally gimme some time 👍

**Comment by seladb** [comment]
@amritanshujohri did you make any progress on that?

**Comment by amritanshujohri** [comment]
Yeah locally we have tested please expect a pr over this weekend.

**Comment by seladb** [comment]
Fixed in PR #551 . Thank you so much @amritanshujohri for your contribution!
